src/OpenTelemetry.Exporter.InfluxDB/InfluxDBMetricsExporterOptions.cs

```diff
@@ -26,22 +26,22 @@ public class InfluxDBMetricsExporterOptions
     /// <summary>
     /// Gets or sets HTTP/S destination for line protocol.
     /// </summary>
-    public Uri Endpoint { get; set; }
+    public Uri Endpoint { get; set; } = null!;
```

Comment:
This seems odd. Do you want to update the types to be nullable instead? For example: `Uri?`, `string?`

Reply:
I intended to mark these bits as non-nullable/required. I'm not sure if this can be elegantly expressed with the available API.

Reply:
This doesn't look correct to me either. One way to do it correctly would be to not use auto-properties:

```csharp
private string bucket = string.Empty;
public string Bucket { 
   get => this.bucket;
   set {
      Guard.ThrowIfNull(value);
      this.bucket = value;
   }
}
```

Not exactly sure what to do with the `Uri` one because you'll need an instance for the default. You could just create some static known default value to check against for "not-set" case. It might be easier just to embrace the nullability though.

Reply:
Interesting suggestion but I'm not sure how does it help to move the check from compile-time to runtime. I'm more inclined to go with the second option.

Reply:
Kind of hard to articulate 😄

In our public APIs we must validate what the user gives us regardless of whatever the nullable annotations are. The annotations are just hints to help analyzers point out potential bugs. Users may not have the analyzers, may not pay attention, or may just do weird things. There is really no compile time or run time enforcement of _anything_ when it comes to nullable reference types. As you have demonstrated here, a user may call your public API with `null!` to quiet the warning and it will compile and run just fine even as you declare the field as non-nullable. If you just rely on the annotations you are likely to run into null reference exceptions. The only thing that will save you in that case is proper good ol' validation of anything that comes into a public API from something external 😄

My best practices would be...

**DO** Decorate our reference types with nullability. This will allow analyzers to point out potential mistakes to developers.
**DO** Validate everything the user gives you regardless of annotations.
**DO NOT** Remove validation of things in public APIs once they are annotated. The annotations are just hints. Users may not be running the analysis or may ignore it, may make mistakes, or may force things using the `!` operator.

Reply:
By second option I had in mind:

> It might be easier just to embrace the nullability though.

Changes applied. :)

Reply:
Ya what you have now is probably the best approach. Embrace that they may be null which will generate warnings in the project where things aren't being checked/validated so it can be made safe.